tests/core/interfaces-gpio-chardev/task.yaml

```diff
@@ -0,0 +1,136 @@
+summary: Ensure that the GPIO chardev interface works.
+
+details: |
+    The gpio-chardev interface allows read/write access to specified GPIO chip lines.
+
+systems: [ubuntu-core-2*]
+
+prepare: |
+    # emulate GPIO chips using gpio-sim module
+    cat << \EOF > gpio-sim.service
+    [Unit]
+    Description=Emulate gpio chips early in boot
+    After=sysinit.target
+    Wants=sysinit.target
+    [Service]
+    Type=oneshot
+    RemainAfterExit=true
+    ExecStart=PWD/mk-sim-chips.sh
```

Comment:
You can expand those right here. Heredocs come in two variants, 'FOO' and FOO. FOO does expand variables.